bcipy/helpers/save.py

```diff
@@ -169,3 +182,35 @@ def save_model(model: SignalModel, path: Union[Path, str]):
         # It supports very large objects and some data format optimizations
         # making it appropriate for signal models.
         pickle.dump(model, file, protocol=4)
+
+
+def save_stimuli_position_info(
+        stimuli_position_info: dict,
+        path: Union[Path, str],
+        screen_info: dict) -> str:
+    """Save stimuli positions and screen info to `path`
+
+   stimuli_position_info: {'A': (0, 0)}
+   screen_info: {'screen_size_pixels': [1920, 1080], 'screen_refresh': 160}
+
+    Parameters
+    ----------
+        stimuli_position_info - stimuli position info to save to json
+        path - path to the file which will be created.
+        screen_info - screen info to save to json
+    """
+    # assert that stimuli_position_info is a dict with string keys and list values
+    assert all(isinstance(key, str) for key in stimuli_position_info.keys()), \
+        'stimuli_position_info keys must be strings'
+    assert list(map(type, (stimuli_position_info[key] for key in stimuli_position_info)))[0] is tuple, \
+        'stimuli_position_info values must be a tuple'
+
+    # assert that screen_info is a dict with at least the key 'screen_resolution'
+    assert 'screen_size_pixels' in screen_info.keys(), \
+        'screen_size_pixels must be a key in screen_info'
+    assert isinstance(screen_info['screen_size_pixels'], list), \
+        'screen_size_pixels must be a list'

```

Comment:
Given the amount of input validation required would it make sense to have some stricter parameter types?

Reply:
That's true with the new typing it should be mostly caught... I will remove some of this (minus key:value assert) and avoid a more explicit data structure for now.